sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp

```diff
@@ -0,0 +1,97 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+// Code generated by Microsoft (R) AutoRest C++ Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/context.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+#include <azure/core/nullable.hpp>
+#include <azure/core/response.hpp>
+#include <azure/core/url.hpp>
+#include <azure/keyvault/administration/dll_import_export.hpp>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+  /**
+   * @brief The type specifier of the value.
+   */
+  class SettingTypeEnum final {
+  public:
+    /**
+     * @brief Default constructor
+     */
+    SettingTypeEnum() = default;
+
+    /**
+     * @brief Constructor
+     * @param value string value
+     */
+    explicit SettingTypeEnum(std::string value) : m_value(std::move(value)) {}
+    bool operator==(const SettingTypeEnum& other) const { return m_value == other.m_value; }
+    bool operator!=(const SettingTypeEnum& other) const { return !(*this == other); }
+
+    /**
+     * @brief Returns value as string
+     *
+     * @returns String value
+     */
+    const std::string& ToString() const { return m_value; }
+    AZURE_SECURITY_KEYVAULT_ADMINISTRATION_DLLEXPORT const static SettingTypeEnum Boolean;
+
+  private:
+    std::string m_value;
+  };
+
+  /**
+   * @brief Update Security Options
+   */
+  struct UpdateSettingOptions final
+  {
+    /**
+     * The value of the pool setting.
+     */
+    std::string Value;
+  };
+
+  /**
+   * @brief Setting Model
+   */
+  struct Setting final
+  {
+    /**
+     * The account setting to be updated.
+     */
+    std::string Name;
+    /**
+     * The value of the pool setting.
+     */
+    std::string Value;
+    /**
+     * The type specifier of the value.
+     */
+    Azure::Nullable<SettingTypeEnum> Type;
+  };
+
+  /**
+   * @brief The settings list result.
+   */
+  struct SettingsListResult final
```

Comment:
Does the postfix `Result` make sense or should it be `SettingsList`?  This is another item that is probably codegen and we need to make sure it follows a consistent pattern and is logical to the customer